src/main/java/makaroshyna/onlinebookstore/mapper/BookMapper.java

```diff
@@ -1,14 +1,32 @@
 package makaroshyna.onlinebookstore.mapper;
 
+import java.util.Set;
+import java.util.stream.Collectors;
 import makaroshyna.onlinebookstore.config.MapperConfig;
 import makaroshyna.onlinebookstore.dto.book.BookDto;
+import makaroshyna.onlinebookstore.dto.book.BookDtoWithoutCategoryIds;
 import makaroshyna.onlinebookstore.dto.book.CreateBookRequestDto;
 import makaroshyna.onlinebookstore.model.Book;
+import makaroshyna.onlinebookstore.model.Category;
+import org.mapstruct.AfterMapping;
 import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+import org.mapstruct.MappingTarget;
 
 @Mapper(config = MapperConfig.class)
 public interface BookMapper {
+    @Mapping(target = "categoryIds", ignore = true)
```

Comment:
setting ignore = true might not be necessary because @AfterMapping will handle setting the categoryIds field regardless. However, if you want to ensure that MapStruct doesn't generate unnecessary mapping code for the categoryIds field, setting ignore = true could be a valid approach